### PR TITLE
Add a feature gate to enable hostDisks

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
This PR adds the feature gate "HostDisk" that has to be enabled to use hostDisks.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
The "HostDisk" feature gate has to be enabled to use hostDisks
```

### CODE REVIEW

File: tests/storage_test.go [hunk 1]
```diff
@@ -402,6 +410,23 @@ var _ = Describe("Storage", func() {
...
+
+					It("Should fail to start a VMI", func() {
+						diskName := "disk-" + uuid.NewRandom().String() + ".img"
+						diskPath := filepath.Join(hostDiskDir, diskName)
+						vmi = tests.NewRandomVMIWithHostDisk(diskPath, v1.HostDiskExistsOrCreate, "")
+						tests.RunVMIAndExpectLaunch(vmi, 30)
+						virtClient, err := kubecli.GetKubevirtClient()
+						Expect(err).ToNot(HaveOccurred())
+						_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+						Expect(err).To(HaveOccurred())
```
Comment (rmohr) [inline tests/storage_test.go:426]: Can we check if the expected cause is included in the error?

File: pkg/virt-config/feature-gates.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ const (
 	SnapshotGate          = "Snapshot"
 	DataVolumesGate       = "DataVolumes"
 	SRIOVGate             = "SRIOVGate"
+	HostDiskGate          = "HostDiskVMIAccess"
```
Comment (rmohr) [inline pkg/virt-config/feature-gates.go:37]: Should we just call it `HostDisk`?

File: tests/storage_test.go (hunk 1 above)
Comment (jean-edouard) [reply tests/storage_test.go:425]: Good point, done.

File: pkg/virt-config/feature-gates.go (hunk 2 above)
Comment (jean-edouard) [reply pkg/virt-config/feature-gates.go]: Renamed.

**Comment by rmohr** [comment]
/lgtm cancel

Removing the PR from the merge pool since it just entered it due to the timewrapper PR.

**Comment by jean-edouard** [comment]
Depends on https://github.com/kubevirt/kubevirt/pull/3542 for stability

**Comment by rmohr** [comment]
@jean-edouard I think the failing tests are failing because they also use hostDisk and the feature gate is not enabled for them.

**Comment by jean-edouard** [comment]
@rmohr good catch, missed one, force-pushed a fix

**Pull request merged**